app/assets/javascripts/components/fight.jsx

```diff
@@ -0,0 +1,231 @@
+        chest: "Hit ches",
+        stomach: "Hit stomach",
+        lowerback: "Hit lowerback",
+        belt: "Hit belt",
+        legs: "Hit legs"
+      },
+      block: {
+        head: "Block head",
+        chest: "Block ches",
+        stomach: "Block stomach",
+        lowerback: "Block lowerback",
+        belt: "Block belt",
+        legs: "Block legs"
+      },
+      demageU: 10,
+      demageB: 13
+    }
+  }
+
+  setAnimationLog(imgPath, log){
+      this.setState({ urlImg: imgPath });
+      this.setState({ hitlog: log });
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+
+    let userH = this.state.userH;
+    let name = event.target.name;
+    let value = event.target.value;
+    userH[name] = value;
+
+    this.setState({userH});
+  }
+
+  botRandHit() {
+    let countSelect = this.child.countOptions()
+    let botH = this.state.botH;
+
+    let attack = Math.floor(Math.random() * countSelect) + 0 ;
+    let block =  Math.floor(Math.random() * countSelect) + 0 ;
+
+    botH[0] = attack;
+    botH[1] = block;
+
+    this.setState({botH})
+  }
+
+
+  checkPlayersMiss(attack, block, player) {
+    if ( attack != block) {
+      if (player == 'Bot') {
+        this.props.bot.hp -= this.state.demageU;
+      }else
+       this.props.user.hp -= this.state.demageU;
+    }
+  }
+
+
+  runFight() {
+    this.buttonEnabled(false);
+    this.botRandHit()
+
+    let attackU = this.state.userH["attackU"]
+    let blockU = this.state.userH["blockU"];
+    let attackB = this.state.botH[0];
+    let blockB = this.state.botH[1];
+
+    this.checkPlayersMiss(attackU, blockB, 'Bot')
+    this.setAnimationLog("/assets/hit-1.gif", "User hit")
+
+    let promise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve(this.buttonEnabled(true));
+      }, 4000);
+    });
+    promise
+    .then(
+      result => {
+        this.checkPlayersMiss(attackB , blockU, 'User')
+        this.setAnimationLog("/assets/hit-2.gif", "Bot hit")
+      }
+    )
+  }
+
+  playerAlive(player) {
+    if (player == 'User') {
+      if (this.props.user.hp <= 0)
+      {
+        this.props.user.hp = 0;
+      }
+    return this.props.user.hp
+      }
+      else if (this.props.bot.hp <= 0)
+      {
+        this.props.bot.hp = 0;
+      }
+    return this.props.bot.hp
+  }
+
+  changeFightProperty() {
+    const user = this.props.user;
+    const bot = this.props.bot;
+    let EXP = 0;
+    let winner = 'BOT';
+
+    if (bot.hp <=0 || user.hp <=90) {
+      switch (true) {
+        case user.hp < 0 && bot.hp < 0:
+          EXP = 5;
+          winner = 'DRAW!. Really? It happens sometimes..';
+          break;
+        case user.hp > bot.hp:
+          EXP = 10;
+          winner = 'YOU';
+          break;
+        default:
+          EXP = 1;
+          break;
+      }
+
+      this.sendExp(EXP).then(() => {
+        this.buttonEnabled(false);
+
+        alert (
+         `The fight has ENDED! And the winner is: ${winner}!` +
```

Comment:
should not be the message like this + User should be redirected automatically after fight is finished.